update appraisal documentation (#1125)
EnergySystemsModellingLab/MUSE_2.0: docs/model/investment.md

### Pre-calculation of metrics for each supply option

- Annualised fixed costs per unit of capacity (\\( AFC_{opt,r} \\)): For new candidates, this is
  their annualised CAPEX plus FOM. For existing assets, the relevant fixed cost is its FOM.

- Calculate net revenue per unit of activity \\(AC_{t}^{NPV} \\) (Tool A):
  \\[
    \begin{aligned}
          AC_{t}^{NPV} = &-cost\_{\text{var}}[t] \\\\

>             &- \sum\_{c} \Big( cost\_{\text{input}}[c] \cdot input\_{\text{coeff}}[c] +

Reviewer: I think maybe the sign is wrong here? Input coefficients are negative, so this is adding (double negative) input costs to the revenue, whereas we should be subtracting. Maybe we should represent this as abs(coeff) like we do in the code?

In general I find these equations thoroughly confusing

### Pre-calculation of metrics for each supply option

- Annualised fixed costs per unit of capacity (\\( AFC_{opt,r} \\)): For new candidates, this is
  their annualised CAPEX plus FOM. For existing assets, the relevant fixed cost is its FOM.

- Calculate net revenue per unit of activity \\(AC_{t}^{NPV} \\) (Tool A):
  \\[
  \begin{aligned}
  AC_{t}^{NPV} = & \quad -cost\_{\text{var}}[t] \\\\
  &- \sum\_{c} \Big( cost\_{\text{input}}[c] \cdot input\_{\text{coeff}}[c] +
  cost\_{\text{output}}[c] \cdot output\_{\text{coeff}}[c] \Big) \\\\
  &+ \sum\_{c} \Big( output\_{\text{coeff}}[c] - input\_{\text{coeff}}[c] \Big)
    \cdot \lambda\_{c,r,t} \\\\
  &+ \sum\_{s,c} in\\_scope[s] \cdot \Big\\{ \\\\
  &\quad \quad (\mu\_{s,c}^{\text{prod}} - cost\_{\text{prod}}[s,c])
    \cdot output\_{\text{coeff}}[c] \\\\
  &\quad \quad + (\mu\_{s,c}^{\text{cons}} - cost\_{\text{cons}}[s,c])
    \cdot input\_{\text{coeff}}[c] \\\\
  &\quad \quad + (\mu\_{s,c}^{\text{net}} - cost\_{\text{net}}[s,c])
    \cdot (output\_{\text{coeff}}[c] - input\_{\text{coeff}}[c]) \\\\
  &\Big\\}
  \end{aligned}
  \\]

- Calculate cost per unit of activity \\( AC_{t}^{LCOX} \\) (Tool B). Note that the commodity

>  of interest (primary output \\( c_{primary} \\)) is excluded from the price term:

Reviewer 2: ```suggestion
  of interest (primary output \\( c_{primary} \\)) is excluded from the price term:
```

>   \\[

Reviewer 2: @Aurashk?

- **Optimise capacity and dispatch to maximise annualised profit:** Solve a small optimisation
  sub-problem to maximise the asset's surplus, subject to its operational rules and the specific

>   demand tranche it is being asked to serve. \\(\varepsilon \approx 1×10^{-14}\\) is added to each

Reviewer 2: Agree with Copilot

#### Tool A: NPV

This method is used when decision rule is single objective and objective is annualised profit for

> agents' serving commodity \\( c \\). It iteratively builds a supply portfolio by selecting

Reviewer 2: ```suggestion
agents serving commodity \\( c \\). It iteratively builds a supply portfolio by selecting
```

- **Optimise capacity and dispatch to maximise annualised profit:** Solve a small optimisation
  sub-problem to maximise the asset's surplus, subject to its operational rules and the specific

>   demand tranche it is being asked to serve. \\(\varepsilon \approx 1×10^{-14}\\) is added to each

Reviewer 2: Oh, if it's negative I feel like we should just set it to `f64::EPSILON`, i.e. use `number.max(f64::EPSILON)`. I think this would be a cleaner approach than just always adding this constant. Maybe this should be done in a separate PR though...

Do you have thoughts @tsmbland?

  - Capacity is constrained to \\( CapMaxBuild \\) for candidates, and to known capacity for
    existing assets.

- **Calculate a profitability index:** This is the total annualised surplus divided by the
 annualised fixed cost.
  \\[
  \text{Profitability Index} =

>   \frac{\sum_t \text{act}_t \times \text{AC}_t^{\text{NPV}}}{\text{AFC} \times \text{cap}}

Reviewer 2: Maybe let's use `\cdot`s for consistency:

```suggestion
  \frac{\sum_t \text{act}_t \cdot \text{AC}_t^{\text{NPV}}}{\text{AFC} \cdot \text{cap}}
```

>   \\]

Reviewer 2: We should also mention that we fall back on comparing the numerator of the PI when AFC is zero.

    each timeslice of the tranche, which adapts based on the commodity's balance level (time slice,
    season, annual).

  - Capacity is constrained up to \\( CapMaxBuild \\) for candidates, and to known capacity for
    existing assets.

  - VoLL variables are active to ensure a feasible solution alongside maximum operation of the
    asset.

- **Calculate a Cost Index Metric:** This is the total annualised cost divided by the annual output.
  \\[
  \text{Cost Index} = \frac{\text{AFC} \times \text{cap}_r + \sum_t act_t
   \times \text{AC}_t^{\text{LCOX}}}{\sum_t act_t}
  \\]

#### Equal-Metric Fallback

If two or more investment options from the same tool have equal metrics, the following tie-breaking
rules are applied in order:

1. Assets which are already commissioned are preferred over new candidate assets.

2. Newer (commissioned later) assets are preferred over older assets.

3. If there is still a tie, the first option in the data structure storing the metrics is selected,

>   which is non-deterministic. A warning is emitted when this occurs.

Reviewer 2: It's not "non-deterministic" -- you get the same result every time you run the program. It is arbitrary, though.

The whitespace could also be tidied.

```suggestion
1. Assets which are already commissioned are preferred over new candidate assets.
2. Newer (commissioned later) assets are preferred over older assets.
3. If there is still a tie, the first option in the data structure storing the metrics is selected,
   which is an arbitrary choice. A warning is emitted when this occurs.
```

## Example: Gas Power Plant

The following is an illustrative example of how the NPV and LCOX approaches work, using a simple
gas combined-cycle power plant as the supply option under consideration.
This example demonstrates the evaluation across two time periods
\\(t_0\\) (peak period) and \\(t_1\\) (off-peak period) with variable operating costs
 \\( cost\_{var}[t] \\) constant in all time periods.

### Model Parameters

#### Asset Parameters
<!-- markdownlint-disable MD013 -->
| Parameter | Notation | Value | Description |
| ----------- | ---------- | ------- | ------------- |
| Primary output (Electricity) | \\( output\_{coeff}[c_{primary}] \\) | 1.0 MWh per unit activity | Main commodity produced |
| By-product output (Waste heat) | \\( output\_{coeff}[c_{heat}] \\) | 0.5 MWh per unit activity | Co-product from generation |
| Input (Natural gas) | \\( input\_{coeff}[c_{gas}] \\) | 2.5 MWh per unit activity | Fuel consumption |

> | Variable operating cost | \\( cost\_{var}[t] \\) | £5/MWh of activity | Operating costs per unit activity |

Author: I did try, but it didn't seem to want to change the file at all, so might have missed something.

> | Variable operating cost | \\( cost\_{var}[t] \\) | £5/MWh of activity | Operating costs per unit activity |

Reviewer 2: It worked for me when I ran `npx markdown-table-formatter docs/model/investment.md`.

Anyways, I've committed the result and pushed it to this branch.

\\[
\max \sum_t act_t \cdot AC_t^{NPV} = act_{t_{0}} \cdot 10 + act_{t_1} \cdot (-10)
\\]

where \\( act_t \\) is the activity (operational level) in each time slice, subject to operational
 constraints and demand requirements.

In this case, the optimiser will prefer to dispatch the asset during \\(t_0\\) (profitable) and
 minimise operation during \\(t_1\\) (unprofitable), subject to technical constraints such as minimum
load requirements.

#### Profitability Index

The profitability index is calculated as:

\\[
\text{PI} = \frac{\sum_t act_t \cdot AC_t^{NPV}}{AFC \times cap}
\\]

Suppose the dispatch optimiser determines \\( act_{t_{0}} = 80 \\) MWh and \\( act_{t_1} = 20 \\)
MWh are the optimal activity levels:

\\[
\begin{aligned}
\text{PI} &= \frac{(80 \times 10) + (20 \times (-10))}{1{,}000 \times 100} \\\\
&= \frac{800 - 200}{100{,}000} \\\\
&= \frac{600}{100{,}000} \\\\
&= 0.006
\end{aligned}
\\]

The profitability index is then compared against all other options to determine which asset provides
 the best return on investment for serving the demand.

### LCOX Approach (Tool B)

#### Net Cost per Unit of Activity

**For \\(t_0\\) (peak period):**

\\[
\begin{aligned}

> AC_{t_{0}}^{LCOX} &= 5 + (2.5 \times 35) - (0.5 \times 25) \\\\

Reviewer 2: ```suggestion
AC\_{t\_{0}}^{LCOX} &= 5 + (2.5 \times 35) - (0.5 \times 25) \\\\
```

  - VoLL variables are active to ensure a feasible solution alongside maximum operation of the
    asset.

- **Calculate a cost index:** This is the total annualised cost divided by the annual output.
  \\[
  \text{Cost Index} = \frac{\text{AFC} \times \text{cap}_r + \sum_t \text{act}_t

>    \times \text{AC}_t^{\text{LCOX}}}{\sum_t \text{act}_t}

Author: should I update the rest of the file to consistently use `\text` as copilot says? then we can make an issue about making sure it's consistent throughout the docs? (Maybe we can assign copilot to do that too :) )

| By-product output (Waste heat) | \\( output\_{coeff}[c_{heat}] \\) | 0.5 MWh per unit activity | Co-product from generation |
| Input (Natural gas) | \\( input\_{coeff}[c_{gas}] \\) | 2.5 MWh per unit activity | Fuel consumption |
| Variable operating cost | \\( cost\_{var}[t] \\) | £5/MWh of activity | Operating costs per unit activity |
<!-- markdownlint-enable MD013 -->

#### Investment Parameters

| Parameter | Notation | Value |
|-----------|----------|-------|
| Annualised fixed cost | \\( AFC\_{opt,r} \\) | £1,000/MW |
| Capacity | \\( cap \\) | 100 MW |

#### Market Prices by Time Period

| Commodity | Notation | t1 (Peak) | t2 (Off-peak) |
|-----------|----------|-----------|---------------|
| Electricity | \\( \lambda\_{c_{elec},r,t} \\) | £90/MWh | £50/MWh |
| Heat | \\( \lambda\_{c_{heat},r,t} \\) | £25/MWh | £15/MWh |
| Natural gas | \\( \lambda\_{c_{gas},r,t} \\) | £35/MWh | £25/MWh |

### NPV Approach (Tool A)

#### Calculate Net Revenue per Unit of Activity

The net revenue calculation follows the general form:

\\[
AC_t^{NPV} = \sum_{c} \Big( output\_{coeff}[c] - input\_{coeff}[c] \Big) \cdot \lambda_{c,r,t} - cost\_{var}[t]
\\]

> 

Author: I removed the general forms

**For t1 (peak period):**

\\[
\begin{aligned}
AC_{t1}^{NPV} &= (1.0 \times 90) + (0.5 \times 25) + (-2.5 \times 35) - 5 \\\\
&= 90 + 12.5 - 87.5 - 5 \\\\
&= \text{£10/MWh}
\end{aligned}
\\]

The asset earns £10 profit for every MWh it operates during peak periods.

**For t2 (off-peak period):**

\\[
\begin{aligned}
AC_{t2}^{NPV} &= (1.0 \times 50) + (0.5 \times 15) + (-2.5 \times 25) - 5 \\\\
&= 50 + 7.5 - 62.5 - 5 \\\\
&= \text{£} -10 \text{/MWh}

> \end{aligned}

Author: Looks ok to me
